docker/dockerfile.merlin

```diff
@@ -96,7 +96,7 @@ RUN pip install betterproto cachetools graphviz nvtx scipy scikit-learn
 RUN pip install tritonclient[all] grpcio-channelz fiddle wandb npy-append-array
 RUN pip install git+https://github.com/rapidsai/asvdb.git@main
 # xgboost and treelite are pinned from tritonserver support
-RUN pip install xgboost==1.6.2 lightgbm treelite==2.3.0 treelite_runtime==2.3.0
+RUN pip install xgboost==1.6.2 lightgbm treelite==2.4.0 treelite_runtime==2.4.0
```

Comment:
since treelite needs to match the Trtion version due to the [serialization compatibility between versions](https://treelite.readthedocs.io/en/latest/notes-on-serialization.html). That suggests that we'll only be able to build against either the old or new versions of triton, not both, in our container builds. 

If we want to continue building against the older version, we may need to add this as a build argument to allow us to parametrize this in our container build.

Reply:
Agreed on the consequences. I don't think we want to keep building against the old version though? This change was already tested out in the prerelease pipelines, so I think we're okay to use it for the 22.12 release. I don't have a strong opinion either way though—this was just me trying to pitch in on addressing version bump issues that would affect the release.

Reply:
In the ideal case yeah we won't need to worry about the old version. If for some reason we run into some issues with the new release and need to switch back to the previous one then this might need to be changed back. We can come back and add a build arg if we need it later.